### PR TITLE
Add objective metric estimation model for speech enhancement

### PR DESCRIPTION
**Pull request by nateanl**

### CODE REVIEW

File: torchaudio/prototype/models/squim/objective.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+    """Build a custome :class:`torchaudio.prototype.models.SQUIM_OBJECTIVE` model.
+
+    Args:
+        feat_dim (int, optional): The feature dimension after Encoder module.
+        win_len (int): Kernel size in the Encoder module.
+        d_model (int): The number of expected features in the input.
+        nhead (int): Number of heads in the multi-head attention model.
+        hidden_dim (int): Hidden dimension in the RNN layer of DPRNN.
+        num_blocks (int): Number of DPRNN layers.
+        rnn_type (str): Type of RNN in DPRNN.
```
Comment (xiaohui-zhang) [inline torchaudio/prototype/models/squim/objective.py:289]: list acceptable candidate arguments?

**Comment by faroit** [comment]
Just saw this PR wondering if this is liked to a submitted paper?

File: torchaudio/prototype/models/squim/objective.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+
+def transform_wb_pesq_range(x: float) -> float:
+    """Map from the P.862 score to the P.862.2 score.
+    Officially, only the metric defined by ITU-T P.862 is called 'PESQ score',
```
Comment (anuragkr90) [inline torchaudio/prototype/models/squim/objective.py:11]: Change in the language of the comment. 

The metric defined by ITU-T P.862 is called often called 'PESQ score', and is defined for narrow-band signals and has a value range of [-0.5, 4.5] exactly. Here, we are using the metric defined by ITU-T P.862.2, commonly known as 'wide-band PESQ' and will be referred to as "PESQ score".

**Comment by nateanl** [comment]
> Just saw this PR wondering if this is liked to a submitted paper?

@faroit Yes, the paper was just accepted by ICASSP. I'll add citation to the code once the proceedings are out.

File: torchaudio/prototype/models/squim/objective.py (hunk 1 above)
Comment (nateanl) [reply torchaudio/prototype/models/squim/objective.py]: addressed in [e906bba](https://github.com/pytorch/audio/pull/3042/commits/e906bba161e67c2917a703b11f26e54707f33b93)

File: torchaudio/prototype/models/squim/objective.py (hunk 2 above)
Comment (nateanl) [reply torchaudio/prototype/models/squim/objective.py]: addressed in [e906bba](https://github.com/pytorch/audio/pull/3042/commits/e906bba161e67c2917a703b11f26e54707f33b93)

File: docs/source/prototype.models.rst [hunk 3]
```diff
@@ -91,3 +91,21 @@ hifigan_vocoder_v3
 ~~~~~~~~~~~~~~~~~~
 
 .. autofunction:: hifigan_vocoder_v3
+
+
+SQUIM_OBJECTIVE
```
Comment (mthrok) [inline docs/source/prototype.models.rst:96]: Could you use autosummary?

**Review by mthrok — APPROVED** [review]

File: docs/source/prototype.models.rst (hunk 3 above)
Comment (nateanl) [reply docs/source/prototype.models.rst:96]: it turns out the code change should also cover other models. Let me keep what it is and create another PR for using autosummary.

**Review by JupiterEthan — COMMENTED** [review]
LGTM
